atc/db/build_factory.go

```diff
@@ -178,7 +178,9 @@ func getBuildsWithDates(buildsQuery, minMaxIdQuery sq.SelectBuilder, page Page,
 
 		defer sinceRow.Close()
 
+		found := false
```

Comment:
Are there any tests to update/add for this or is it more of an optimization that's already technically exercised by existing tests and hard to test in isolation?

Would you mind splitting it out into a separate commit, just so it's clear that it's not strictly part of the pipeline collector logic?

Reply:
Sure, I make a separate PR to fix this bug.